gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +721,671 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+                 "No Radii keyword available, metadata will not contain DATUM information.");
+            status = 0;
+        } else {
+            fits_read_key(hFITS, TDOUBLE, "C_RADIUS", &cRadius, nullptr, &status);
+            if( aRadius != cRadius )
+            {
+                invFlattening = aRadius / ( aRadius - cRadius );
+            }
+        }
+
+        /* Waiting for linear keywords standardization only deg ctype are used */
+        /* Check if WCS are there */
+        fits_read_key(hFITS, TSTRING, "CTYPE1", ctype, nullptr, &status);
+        if ( !status ) {
+            /* Check if angular WCS are there */
+            if ( strstr(ctype, "LN") )
+            {
+                /* Reading reference points */
+                fits_read_key(hFITS, TDOUBLE, "CRPIX1", &crpix1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRPIX2", &crpix2, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL1", &crval1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL2", &crval2, nullptr, &status);
+                /* Check for CDELT and PC matrix representation */
+                fits_read_key(hFITS, TDOUBLE, "CDELT1", &cdelt1, nullptr, &status);
+                if ( ! status ) {
+                    fits_read_key(hFITS, TDOUBLE, "CDELT2", &cdelt2, nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_1", &pc[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_2", &pc[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_1", &pc[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_2", &pc[3], nullptr, &status);
+                    cd[0] = cdelt1 * pc[0];
+                    cd[1] = cdelt1 * pc[1];
+                    cd[2] = cdelt2 * pc[2];
+                    cd[3] = cdelt2 * pc[3];
+                    status = 0;
+                } else {
+                    /* Look for CD matrix representation */
+                    fits_read_key(hFITS, TDOUBLE, "CD1_1", &cd[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD1_2", &cd[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_1", &cd[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_2", &cd[3], nullptr, &status);
+                }
+
+                double radfac = DEG2RAD * aRadius;
+
+                adfGeoTransform[1] = cd[0] * radfac;
+                adfGeoTransform[2] = cd[1] * radfac;
+                adfGeoTransform[4] = cd[2] * radfac;
+                adfGeoTransform[5] = - cd[3] * radfac ;
+                if ( crval1 > 180. ) {
+                    crval1 = crval1 - 180.;
+                }
+                adfGeoTransform[0] = crval1 * radfac - adfGeoTransform[1] *
+                                                    (crpix1-0.5);
+                adfGeoTransform[3] = crval2 * radfac - adfGeoTransform[5] *
```

Comment:
`crpix1-.05` looks like rounding, but this is double precision, so maybe a comment about what's up?

Reply:
@abellgithub , you're right... this is not clear. I've added a comment in the code about that.
Using the GeoTIFF nomenclature, FITS raster representation is a "PixelIsPoint" Raster Space.
Then the upper-left corner of the image is represented by (-0.5, -0.5) coordinates, that's why I'm subtracting 0.5 in transformations.

Reply:
The pixel coordinate system trips us up every time. I forget is FITS really (-0.5, -0.5)?
https://github.com/OSGeo/gdal/issues/735#issuecomment-403089802

Reply:
> Then the upper-left corner of the image is represented by (-0.5, -0.5) coordinates, that's why I'm subtracting 0.5 in transformations.

Sorry, my bad, upperleft corner is (0.5, 0.5) ... resulting in this particular transformation in subtracting .5

Reply:
> The pixel coordinate system trips us up every time. I forget is FITS really (-0.5, -0.5)?

Finally found the ref [Greisen Calabretta, 2002](https://www.aanda.org/articles/aa/full/2002/45/aah3859/aah3859.html) paragraph 2.1.4 "Additional points"